### PR TITLE
Get grpc from nix on unix

### PR DESCRIPTION
**Pull request by cocreature**
The one from Bazel seems to cause linking issues when trying to run
things in GHCi. I’ve spent some time trying to use rules_foreign_cc to
build gRPC using CMake but decided that for my own sanity it’s better
to not pursue that further.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags, if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: bazel-haskell-deps.bzl [hunk 1]
```diff
@@ -260,6 +260,22 @@ haskell_library(
         urls = ["https://github.com/digital-asset/ghcide/archive/%s.tar.gz" % GHCIDE_REV],
     )
 
+    # The Bazel-provided grpc libs cause issues in GHCi so we get them from Nix on Linux and MacOS.
+    # We keep them in
```
Comment (aherrmann-da) [inline bazel-haskell-deps.bzl]: Is that line a left-over.

File: bazel-haskell-deps.bzl [hunk 2]
```diff
@@ -260,6 +260,22 @@ haskell_library(
...
+    deps = "[\":grpc\", \":libgpr\"]" if is_windows else "[\"@grpc_nix//:grpc_lib\"]"
```
Comment (aherrmann-da) [inline bazel-haskell-deps.bzl]: Not important, but you can also use `'[":grpc", ...]'` to avoid some escaping.

File: build.sh [hunk 3]
```diff
@@ -32,7 +32,13 @@ bazel test -j 200 //... --test_tag_filters "$tag_filter" --experimental_executio
 # Make sure that Bazel query works.
 bazel query 'deps(//...)' > /dev/null
 # Check that we can load damlc in ghci
-da-ghci --data yes //:repl -e '()'
+GHCI_SCRIPT=$(mktemp)
+cat <<EOF > $GHCI_SCRIPT
+:main --help
+:quit
+EOF
+echo $GHC_
```
Comment (aherrmann-da) [inline build.sh]: Where is `$GHC_` defined?

Comment (cocreature) [reply build.sh]: never, removed :slightly_smiling_face:

File: bazel-haskell-deps.bzl (hunk 2 above)
Comment (cocreature) [reply bazel-haskell-deps.bzl]: Ah good point, thanks! Changed.

File: bazel-haskell-deps.bzl (hunk 1 above)
Comment (cocreature) [reply bazel-haskell-deps.bzl]: Yep, removed. Thanks!

File: bazel-haskell-deps.bzl [hunk 4]
```diff
@@ -260,6 +260,21 @@ haskell_library(
         urls = ["https://github.com/digital-asset/ghcide/archive/%s.tar.gz" % GHCIDE_REV],
     )
 
+    # The Bazel-provided grpc libs cause issues in GHCi so we get them from Nix on Linux and MacOS.
+    deps = '[":grpc", ":libgpr"]" if is_windows else "["@grpc_nix//:grpc_lib"]'
```
Comment (aherrmann-da) [inline bazel-haskell-deps.bzl]:
```suggestion
    deps = '[":grpc", ":libgpr"]' if is_windows else '["@grpc_nix//:grpc_lib"]'
```

File: build.sh [hunk 5]
```diff
@@ -32,7 +32,12 @@ bazel test -j 200 //... --test_tag_filters "$tag_filter" --experimental_executio
 # Make sure that Bazel query works.
 bazel query 'deps(//...)' > /dev/null
 # Check that we can load damlc in ghci
-da-ghci --data yes //:repl -e '()'
+GHCI_SCRIPT=$(mktemp)
+cat <<EOF > $GHCI_SCRIPT
+:main --help
+:quit
+EOF
+da-ghci --data yes //:repl -ghci-script $GHCI_SCRIPT -e '()'
```
Comment (SamirTalwar) [inline build.sh]: You might be able to get away with something like this instead:

```
da-ghci --data yes //:repl -ghci-script <(echo ':main --help' && echo ':quit') -e '()'
```

Comment (SamirTalwar) [reply build.sh]: Or use `echo -e ':main --help\n:quit'`, but I think that's harder to read.

Comment (cocreature) [reply build.sh]: tbh I find both harder to read.

Comment (SamirTalwar) [reply build.sh]: Fair. I think I'm a bit allergic to `mktemp` without cleanup.

Comment (aherrmann-da) [reply build.sh]: @SamirTalwar Good point. @cocreature You could add `trap EXIT "rm -f $GHCI_SCRIPT"` below `mktemp` for cleanup (if that's the right spelling).

Comment (SamirTalwar) [reply build.sh]: [not important]

You could also do:

```shell
function ghci_script {
  echo <<EOF
:m DA.Cli.Damlc
:main --help
:quit
EOF
}

da-ghci --data yes //:repl -ghci-script <(ghci_script) -e '()'
```

Or you could just check this file in somewhere and use it. 🙂

Comment (cocreature) [reply build.sh]: I actually tried using a temporary file with `<` and for some reason `ghci` didn’t seem to like that and I don’t really want to spend my time debugging this :slightly_smiling_face:

**Pull request merged**